File: Euclid/Messages/Incoming/Move.cs
using Euclid.Game;
using Euclid.Network.Streams;
using System;

namespace Euclid.Messages.Incoming
{
    class Move : IMessageEvent
    {
        public void Handle(Player player, Request request)
        {

            if (!player.RoomEntity.WalkingAllowed)
                return;

            int x = int.Parse(request.GetArgument(0));
            int y = int.Parse(request.GetArgument(1));

            Console.WriteLine(new Position(x, y).GetTile(player.RoomEntity.Room).HighestItem != null ? new Position(x, y).GetTile(player.RoomEntity.Room).HighestItem.Definition.Data.Sprite : null);

            player.RoomEntity.Move(x, y);
        }
    }
}

File: euclid/src/main/java/euclid/messages/incoming/Move.java
package euclid.messages.incoming;

import euclid.game.Player;
import euclid.messages.IMessageEvent;
import euclid.network.streams.Request;

public class Move implements IMessageEvent {

    @Override
    public void handle(Player player, Request request) {
        if (!player.getRoomEntity().isWalkingAllowed()) {
            return;
        }

        int x = Integer.parseInt(request.getArgument(0));
        int y = Integer.parseInt(request.getArgument(1));

        player.getRoomEntity().move(x, y);
    }
}
